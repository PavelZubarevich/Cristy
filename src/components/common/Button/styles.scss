@import '../../../styles/vars.scss';

.button {
  background-color: transparent;
  border: 0;
  font-family: "Lora";
  text-transform: capitalize;
  cursor: pointer;
  transition: 0.3s;

  &_sm {
    font-size: 14px;
    font-weight: 400;
  }

  &_lg {
    font-size: 16px;
    font-weight: 700;
  }

  &_contained {
    padding: 15px 35px;

    &_primary {
      background-color: $primary;
      color: #fff;
    }

    &:hover {
      opacity: 0.75;
    }
  }

  &_text {
    &_primary {
      color: $primary;
    }

    &_primaryText {
      color: $primaryText;
    }

    &:hover {
      opacity: 0.75;
    }
  }

  &_icon {
    width: 50px;
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;

    &_primary {
      background-color: $primary;

      &>svg {
        color: #fff;
        font-size: 22px;
      }
    }

    &_white {
      background-color: #fff;

      &>svg {
        color: $primaryDarkText;
        font-size: 22px;
      }
    }

    &:hover {
      opacity: 0.75;
    }
  }

  &_outlined {
    padding: 15px 35px;
    border: 2px solid $primary;
    color: $primaryDarkText;

    &:hover {
      background-color: $primary;
      color: #fff;
    }
  }
}