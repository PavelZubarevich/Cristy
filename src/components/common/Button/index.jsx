import PropTypes from 'prop-types';
import './styles.scss';

function Button({
  text, variant, color, className, size, onClick,
}) {
  return (
    <button type="button" onClick={onClick} className={`button button_${size} button_${variant} button_${variant}_${color} ${className}`}>{text}</button>
  );
}

export default Button;

Button.propTypes = {
  text: PropTypes.string.isRequired,
  variant: PropTypes.oneOf(['contained', 'text']),
  color: PropTypes.oneOf(['primary', 'primaryText']),
  className: PropTypes.string,
  size: PropTypes.oneOf(['sm', 'lg']),
  onClick: PropTypes.func,
};

Button.defaultProps = {
  variant: 'primary',
  color: 'primary',
  className: '',
  size: 'lg',
  onClick: () => {},
};
