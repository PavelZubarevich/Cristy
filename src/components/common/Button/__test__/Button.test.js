import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '..';

describe('Button', () => {
  describe('button contained', () => {
    it('should render large primary button', () => {
      render(<Button
        variant="contained"
      />);
      expect(document.body).toMatchSnapshot();
    });
    it('should render small primary button', () => {
      render(<Button
        variant="contained"
        size="sm"
      />);
      expect(document.body).toMatchSnapshot();
    });
  });
  describe('button text', () => {
    it('should render large primary button', () => {
      render(<Button
        variant="text"
      />);
      expect(document.body).toMatchSnapshot();
    });

    it('should render small primary button', () => {
      render(<Button
        variant="text"
        size="sm"
      />);
      expect(document.body).toMatchSnapshot();
    });

    it('should render large primaryText button', () => {
      render(<Button
        variant="text"
      />);
      expect(document.body).toMatchSnapshot();
    });

    it('should render small primaryText button', () => {
      render(<Button
        variant="text"
        size="sm"
      />);
      expect(document.body).toMatchSnapshot();
    });
  });
  describe('button icon', () => {
    it('should render primary button', () => {
      render(<Button
        variant="icon"
      />);
      expect(document.body).toMatchSnapshot();
    });

    it('should render white button', () => {
      render(<Button
        variant="icon"
        color="white"
      />);
      expect(document.body).toMatchSnapshot();
    });
  });
  describe('button outlined', () => {
    it('should render button', () => {
      render(<Button
        variant="outlined"
      />);
      expect(document.body).toMatchSnapshot();
    });
  });

  it('should handle click', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick} />);
    userEvent.click(screen.getByRole('button'));
    expect(handleClick).toBeCalledTimes(1);
  });

  it('should be default onCkick handler', () => {
    expect(Button.defaultProps.onClick()).toBe(undefined);
  });
});
