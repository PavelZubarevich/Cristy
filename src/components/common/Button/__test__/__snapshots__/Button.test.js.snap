// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button button contained should render large primary button 1`] = `
<body>
  <div>
    <button
      class="button button_lg button_contained button_contained_primary"
      type="button"
    />
  </div>
</body>
`;

exports[`Button button contained should render small primary button 1`] = `
<body>
  <div>
    <button
      class="button button_sm button_contained button_contained_primary"
      type="button"
    />
  </div>
</body>
`;

exports[`Button button icon should render primary button 1`] = `
<body>
  <div>
    <button
      class="button button_lg button_icon button_icon_primary"
      type="button"
    />
  </div>
</body>
`;

exports[`Button button icon should render white button 1`] = `
<body>
  <div>
    <button
      class="button button_lg button_icon button_icon_white"
      type="button"
    />
  </div>
</body>
`;

exports[`Button button outlined should render button 1`] = `
<body>
  <div>
    <button
      class="button button_lg button_outlined button_outlined_primary"
      type="button"
    />
  </div>
</body>
`;

exports[`Button button text should render large primary button 1`] = `
<body>
  <div>
    <button
      class="button button_lg button_text button_text_primary"
      type="button"
    />
  </div>
</body>
`;

exports[`Button button text should render large primaryText button 1`] = `
<body>
  <div>
    <button
      class="button button_lg button_text button_text_primary"
      type="button"
    />
  </div>
</body>
`;

exports[`Button button text should render small primary button 1`] = `
<body>
  <div>
    <button
      class="button button_sm button_text button_text_primary"
      type="button"
    />
  </div>
</body>
`;

exports[`Button button text should render small primaryText button 1`] = `
<body>
  <div>
    <button
      class="button button_sm button_text button_text_primary"
      type="button"
    />
  </div>
</body>
`;
